File: instaflux/src/debug/java/org/wordpress/android/fluxc/instaflux/AppComponentDebug.java
package org.wordpress.android.fluxc.instaflux;

import org.wordpress.android.fluxc.module.AppContextModule;
import org.wordpress.android.fluxc.module.DatabaseModule;
import org.wordpress.android.fluxc.module.OkHttpClientModule;
import org.wordpress.android.fluxc.module.ReleaseNetworkModule;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {
        AppContextModule.class,
        AppConfigModule.class,
        OkHttpClientModule.class,
        InterceptorModule.class,
        ReleaseNetworkModule.class
})
public interface AppComponentDebug extends AppComponent {}

File: example/src/debug/java/org/wordpress/android/fluxc/example/di/AppComponentDebug.kt
package org.wordpress.android.fluxc.example.di

import android.app.Application
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjectionModule
import org.wordpress.android.fluxc.module.DatabaseModule
import org.wordpress.android.fluxc.module.OkHttpClientModule
import org.wordpress.android.fluxc.module.ReleaseNetworkModule
import javax.inject.Singleton
import org.wordpress.android.fluxc.di.WCDatabaseModule

@Singleton
@Component(modules = [
        AndroidInjectionModule::class,
        ApplicationModule::class,
        AppConfigModule::class,
        InterceptorModule::class,
        OkHttpClientModule::class,
        ReleaseNetworkModule::class,
        MainActivityModule::class,
        DatabaseModule::class,
        WCDatabaseModule::class,
        WCOrderListActivityModule::class])
interface AppComponentDebug : AppComponent {
    @Component.Builder
    interface Builder : AppComponent.Builder {
        @BindsInstance
        override fun application(application: Application): AppComponentDebug.Builder
    }
}
